src/mmw/js/src/data_catalog/models.js

```diff
@@ -396,6 +474,115 @@ var PopoverControllerModel = Backbone.Model.extend({
     }
 });
 
+var CuahsiValue = Backbone.Model.extend({
+    defaults: {
+        source_id: '',
+        source_code: '',
+        quality_control_level_code: '',
+        value: null,
+        datetime: '',
+        date_time_utc: '',
+        time_offset: '',
+    }
+});
+
+var CuahsiValues = Backbone.Collection.extend({
+    model: CuahsiValue,
+});
+
+var CuahsiVariable = Backbone.Model.extend({
+    url: '/bigcz/values',
+
+    defaults: {
+        id: '',
+        name: '',
+        units: '',
+        concept_keyword: '',
+        speciation: '',
+        sample_medium: '',
+        wsdl: '',
+        site: '',
+        values: null, // CuahsiValues Collection
+        most_recent_value: null,
+        begin_date: '',
+        end_date: '',
+        error: null,
+    },
+
+    initialize: function() {
+        this.set('values', new CuahsiValues());
+    },
+
+    search: function(from, to) {
+        var self = this,
+            begin_date = moment(this.get('begin_date')),
+            end_date = moment(this.get('end_date')),
+            params = {
+                catalog: 'cuahsi',
+                wsdl: this.get('wsdl'),
+                site: this.get('site'),
+                variable: this.get('id'),
+            };
+
+        // If neither from date nor to date is specified, set time interval
+        // to be either from begin date to end date, or 1 week up to end date,
+        // whichever is shorter.
+        if (!from || moment(from).isBefore(begin_date)) {
+            if (end_date.diff(begin_date, 'weeks', true) > 1) {
+                params.from_date = end_date.subtract(7, 'days');
+            } else {
+                params.from_date = begin_date;
+            }
+        } else {
+            params.from_date = moment(from);
+        }
+
+        if (!to || moment(to).isAfter(end_date)) {
+            params.to_date = end_date;
+        } else {
+            params.to_date = moment(to);
+        }
+
+        params.from_date = params.from_date.format(DATE_FORMAT);
+        params.to_date = params.to_date.format(DATE_FORMAT);
+
+        this.set('error', null);
+
+        return this.fetch({
+                data: params,
+                processData: true,
+            })
+            .fail(function(_, status) {
+                self.set('error', 'Error ' + status + ' during fetch');
```

Comment:
This ends up displaying a little funny:
![screen shot 2017-10-09 at 4 41 37 pm](https://user-images.githubusercontent.com/7633670/31359685-045f223e-ad19-11e7-9f90-e98eea7c990d.png)

Maybe a colon would help?

Reply:
Hmm I thought the `error` was an HTTP code. Will reformat.